src/_P016_IR.ino

```diff
@@ -426,7 +426,7 @@ boolean Plugin_016(byte function, struct EventStruct *event, String &string)
           }
           char strCode[P16_Cchars];
           if (safe_strncpy(strCode, resultToHexidecimal(&results), P16_Cchars)) {
-            iCode += hexToUL(strCode);                               // Bits 21-0 for code
+            iCode += strtoul(strCode, 0, 16);                               // Bits 21-0 for code
```

Comment:
What's wrong with the hexToUL code?
That function uses `strtoul`

Reply:
Didn't investigate deeply, but with a char array for input, it seems to not scan the last byte. Please check the linked issue comments. This fixed that.
May be that changing that array to a String is a better way to fix it, can't see yet why a char array was used in the first place.

Reply:
> May be that changing that array to a String is a better way to fix it, can't see yet why a char array was used in the first place.

Not zero-terminated would be my guess...